### DISCUSSION TITLE
@nx/js:tsc "external" configuration deprecated

### DISCUSSION THREAD

**Discussion by danielsharvey** [nrwl/nx#26439] [Feature Requests]
The `external` configuration for the `@nx/js:tsc` executor is marked as deprecated. It was marked that way in a seemingly unrelated PR https://github.com/nrwl/nx/pull/22856.

The discussion:

> https://github.com/nrwl/nx/pull/22856#issuecomment-2069470116 @jaysoo - I've been following the `external` option with interest, sad to see it's been deprecated. Any chance you can give me some insight into the reasons for discontinuing it? Thanks!

> https://github.com/nrwl/nx/pull/22856#issuecomment-2113235303 @wjehring It is going against what TSC/SWC are supposed to do, just simply transpile files. The experimental feature has proven tricky to support in different scenarios, and we'd rather focus our attention elsewhere.
> 
> Feel free to open a discussion about it in the the Discussions tab, maybe we can address the problems properly without the current unstable feature.

A related issue from my perspective is the recently added declarations/typings support added to the `@nx/esbuild/esbuild` executor (see https://github.com/nrwl/nx/pull/21084 and https://github.com/nrwl/nx/issues/20688).

Under the hood, the esbuild declarations support is achieved using tsc, and as it currently standard the `external` configuration provided to esbuild is not mirrored in the declarations build by the executor using tsc (a limitation that can result in inconsistent between the esbuild output and declarations output).

I'm keen to know the Nx teams thoughts on the direction moving forward here. On one hand, we have external support being marked as deprecated on the other we have new features in the "tsc" space.

To my mind, tsc support (and generating declarations using tsc) is intimately connected with the library dependency graphs that are core to Nx.

Thanks!

**Comment by rachirib** [comment]
Hi community, 🆘  What's the alternative solution about this? 

I have a nx workspace `19.x` monorepo that have a great amount of nestjs applications (`tsc`) with some common libraries that have being configured with `tsc` build as well.

Currently everything works as I have the `external: none` set so it copies all the monorepo common libs inline nicely, though as soon as this is removed, breaks apart.

Does this mean that we need to bundle our apps and libs with a different than `tsc or swc` if monorepo common libs are used? Any doco that can lead me into understanding the migration path would be greatly appreciated.

**Reply by rachirib** [reply]
Found something helpful for anyone getting here as well
https://github.com/nrwl/nx/discussions/27150#discussioncomment-10161002
